src/app/dataset-view/additional-components/overview-component/components/readme-section/readme-section.component.ts

```diff
@@ -31,9 +40,9 @@ export class ReadmeSectionComponent extends BaseComponent implements OnInit {
     private datasetCommitService = inject(DatasetCommitService);
     private loggedUserService = inject(LoggedUserService);
 
-    public ngOnInit(): void {
-        if (this.currentReadme) {
-            this.readmeState = this.currentReadme;
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.currentReadme && changes.currentReadme.currentValue !== changes.currentReadme.previousValue) {
+            this.readmeState = changes.currentReadme.currentValue as string;
         }
     }
```

Comment:
What happens if the dataset switch happens while I"m editing the readme?

Reply:
When switching datasets, the editing mode on the other dataset is saved